projects/views.py

```diff
@@ -519,16 +519,60 @@ def program_delete(request, program_id=0):
             project.active = False
             project.save()
         program.save()
-    else:
-        # Unshare
-        shared_resource = program.shared.filter(matched_user_id=request.user.id)
-        shared_resource.delete()
-
 
     return JsonResponse({
         'status': 'success'
     })
 
+
+@login_required
+def program_unshare(request, program_id=0):
+    message = None
+    status = None
+    status_text = None
+    redirect_url = None
+
+    try:
+        program = Program.objects.get(id=program_id)
+
+        owner = str(program.owner.id)
+        req_user = str(request.user.id)
+        # If a user_id wasn't provided, this is a user asking to remove themselves from a cohort
+        unshare_user = str(request.POST.get('user_id') or request.user.id)
+
+        # You can't remove someone from a program if you're not the owner,
+        # unless you're removing yourself from someone else's program
+        if req_user != owner and req_user != unshare_user:
+            raise Exception('Cannot make changes to sharing on a program if you are not the owner.')
```

Comment:
This was intentional, mostly due to 'security through obscurity'. A user shouldn't be trying to alter an entity which isn't theirs, but we do want to log it. That said, the log message should certainly be more specific to 'someone is trying to alter something they're  no an owner of'.